qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -31,12 +35,14 @@ def __init__(self):
         self._src = None
         self._dst = None
 
-    def encode(self, op: QuadraticProgram, penalty_factor: float = 1e5,
+    def encode(self, op: QuadraticProgram, auto_penalty: bool = True, penalty_factor: float = 1e5,
```

Comment:
Wouldn't it be simpler to have the `penalty_factor` an Optional float and default to None whereupon it would do an automatic penalty value computation. If an explicit value is provided then that would be used of course instead of any automatically computed value.